devtools/__init__.py

```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python3
+
+import sys, os, subprocess, shlex, typing
+from typing import Mapping, Dict, Optional
+
+GITHUB_ACTIONS = bool(os.environ.get('GITHUB_ACTIONS'))
+OFFICIAL_CONTAINER_REPO = 'ghcr.io/evalf/nutils'
+
+if GITHUB_ACTIONS:
+
+  from . import _log_gha as log
+
+  orig_excepthook = sys.excepthook
+
+  def new_excepthook(exctype, value, tb):
+    if exctype == Error:
+      log.error(*value.args)
+    else:
+      orig_excepthook(exctype, value, tb)
+
+  sys.excepthook = new_excepthook
+
+else:
+
+  from . import _log_default as log
+
+class Error(Exception): pass
```

Comment:
For simplicity we could consider removing `Error` and use `sys.exit` instead, catching `SystemExit` in `new_excepthook`. This also removes tracebacks in the non-gha scenario for valid failure paths.

Reply:
I'll replace `raise Error` with `raise SystemExit` if that's OK.

Reply:
sure, I believe there's no diference